packages/react/runtime/__test__/renderToOpcodes.test.jsx

```diff
@@ -4,21 +4,54 @@ import { Component, createContext, Fragment } from 'preact';
 import { useMemo, useState } from 'preact/hooks';
 import { afterEach, beforeAll, beforeEach, describe, expect, it, vi } from 'vitest';
 
-import { elementTree } from './utils/nativeMethod';
+import { elementTree, waitSchedule } from './utils/nativeMethod';
 import { globalEnvManager } from './utils/envManager';
 import { setupDocument } from '../src/document';
 import { renderOpcodesInto } from '../src/opcodes';
 import renderToString from '../src/renderToOpcodes';
 import { setupPage, SnapshotInstance, snapshotInstanceManager } from '../src/snapshot';
 import { createElement, cloneElement } from '../lepus';
+import { Suspense } from 'preact/compat';
+import { Deferred } from '../src/renderToOpcodes/utils';
+
+const defaultChildren = <text>it works</text>;
+
+// `createSuspender()` is forked from `preact` unit test
+export function createSuspender() {
+  const deferred = new Deferred();
```

Comment:
Since we are enforcing using node 22 and above, maybe we could use `Promise.withResolvers` instead.